### ISSUE TITLE
[Bug] Strange Expander behaviour

### ISSUE THREAD

**Issue by ColeXm** [xamarin/XamarinCommunityToolkit#868]
### Description

Trying to use Expander in our app and have strange issue - content view shrinks with each expand/collapse. So when expanded first time it shows content properly and then with each collapse/expand content height gets smaller until completely disappears.

Expander is inside StackLayout, so there is Expander and ListView controls. If I remove ListView then it works fine. Even if ListView is empty it still shrinks.

Tested a bit more and found that it's a Grid inside DataTemplate which is causing the problem, remove grid and it works fine. 

### Code to Reproduce

```
 <StackLayout Orientation="Vertical">
        <xct:Expander Margin="10,0,0,0">
            <xct:Expander.Header>
                <Grid>
                    <Label Text="Total" FontAttributes="Bold"/>
                </Grid>
            </xct:Expander.Header>
            <xct:Expander.ContentTemplate>
                <DataTemplate>
                    <Grid Padding="10">
                        <Grid.ColumnDefinitions>
                            <ColumnDefinition Width="Auto" />
                            <ColumnDefinition Width="Auto" />
                        </Grid.ColumnDefinitions>
                        <Label Text="Stock Total" FontAttributes="Bold" />
                        <Label Text="123456" FontAttributes="Bold" Grid.Column="1" />
                    </Grid>
                </DataTemplate>
            </xct:Expander.ContentTemplate>
        </xct:Expander>

        <ListView>
        </ListView>
    </StackLayout>

```

### Version information

Xamarin.Forms 5.0.0.1874

### Screenshots

![111221](https://user-images.githubusercontent.com/2161256/107310933-64ad1400-6ac8-11eb-99c3-bfbda5e5e757.gif)

### Workaround

Replacing the Grid with StackLayout solves the problem.

**Comment by jfversluis** [comment]
@ColeXm as mentioned by Javier, the Expander lives with us now. Could you please install the Xamarin Community Toolkit and check if this is still an issue as there has some work been done in the meanwhile?

**Comment by AndreiMisiukevich** [comment]
> Tested a bit more and found that it's a Grid inside DataTemplate which is causing the problem, remove the grid and it works fine

I guess, the issue resolved :) right? No need to use greed as ```<Grid></Grid>``` without specifying rows/cols.

**Comment by ColeXm** [comment]
@jfversluis  Yes, problem persists,  actually I'm testing this based on the sample : https://docs.microsoft.com/en-us/samples/xamarin/xamarin-forms-samples/userinterface-expanderdemos/.

**Comment by CliffAgius** [comment]
I can confirm this bug still exists and both the suggested fixes work so either use a StackLayout or set the Grid.Rowdefinitions to Auto for both the Header and the ContentTemplate and the bug disappears.

**Comment by silviuo** [comment]
Hi @ColeXm, do you mind renaming the issue to something pertinent, like what you said in your description - "content view shrinks with each expand/collapse"? Strange behaviour doesn't say anything. I had this exact issue and I fell lucky I found this thread, just because I kept going through everything related to expanders. Thanks!
